src/NewTwitchApi/Resources/EventSubApi.php

```diff
@@ -0,0 +1,286 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NewTwitchApi\Resources;
+
+use GuzzleHttp\Exception\GuzzleException;
+
+class EventSubApi extends AbstractResource
+{
+    /**
+     * @throws GuzzleException
+     * @link https://dev.twitch.tv/docs/eventsub
+     */
+    private function subscribe(string $type, string $version, array $condition, string $callback, string $bearer, string $secret): void
+    {
+        $bodyParams = [];
+
+        $bodyParams[] = ['key' => 'type', 'value' => $type];
+        $bodyParams[] = ['key' => 'version', 'value' => $version];
+        $bodyParams[] = ['key' => 'condition', 'value' => $condition];
+        $bodyParams[] = [
+            'key' => 'transport',
+            'value' => [
+                'method' => 'webhook',
+                'callback' => $callback,
+                'secret' => $secret,
+            ],
+        ];
+
+        $this->postApi('eventsub/subscriptions', $bearer, [], $bodyParams);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelupdate
+     */
+    public function subscribeToChannelUpdate(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribe(
+            'channel.update',
+            '1',
+            ['broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelfollow
+     */
+    public function subscribeToChannelFollow(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribe(
+            'channel.follow',
+            '1',
+            ['broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelsubscribe
+     */
+    public function subscribeToChannelSubscribe(string $twitchId, string $callback, string $bearer, string $secret): void
```

Comment:
- Bearer should always be the first parameter
- Since Secrets are required for all Subscriptions, put this directly after the Bearer
- Since Callbacks are required for all Subscriptions, put this after the Secret